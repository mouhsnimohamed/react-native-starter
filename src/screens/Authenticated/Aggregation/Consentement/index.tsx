import React from 'react';
import { StackNavigationProp } from '@react-navigation/stack';
import { RootStackParamList } from 'navigations/RootNavigation';
import {
  ActionContainer,
  AppBaseView,
  BorderView,
  PrimaryButton,
  PrimaryText,
  Root,
} from 'components';
import { ScrollView } from 'react-native-gesture-handler';
import { theme } from 'components/styles/theme';
import i18n from 'translations';
import styled from 'styled-components/native';
import assets from 'assets/svg';
import addAccessibilityLabel from 'utils/addAccessibilityLabel';
import { useDispatch } from 'react-redux';
import { getBanksAction } from 'modules/aggregation/aggregation.actions';
interface ConsentementProps {
  navigation: StackNavigationProp<RootStackParamList, 'Consentement'>;
}
const Consentement: React.FC<ConsentementProps> = ({ navigation }) => {
  const dispatch = useDispatch();
  return (
    <Root>
      <ScrollView
        keyboardShouldPersistTaps="always"
        keyboardDismissMode="on-drag">
        <AppBaseView justifyContent="flex-start">
          <PrimaryText
            style={{ marginTop: 20, marginBottom: 20 }}
            size={theme.fontSize.medium}>
            {i18n.t('aggregations.consentement.firstTitle')}
          </PrimaryText>
          <CustomBorderView
            radius={10}
            pTop={20}
            pLeft={20}
            pRight={20}
            pBottom={10}>
            <Item>
              <PrimaryText semiBold isPrimary align="left">
                {i18n.t('aggregations.consentement.item_1')}
              </PrimaryText>
            </Item>
            <Item>
              <IconContainer>
                <assets.GREEN_ARROW />
              </IconContainer>
              <PrimaryText isPrimary align="left">
                {i18n.t('aggregations.consentement.item_2')}
              </PrimaryText>
            </Item>
            <Item>
              <IconContainer>
                <assets.GREEN_ARROW />
              </IconContainer>
              <PrimaryText isPrimary align="left">
                {i18n.t('aggregations.consentement.item_3')}
              </PrimaryText>
            </Item>
            <Item>
              <IconContainer>
                <assets.GREEN_ARROW />
              </IconContainer>
              <PrimaryText isPrimary align="left">
                {i18n.t('aggregations.consentement.item_4')}
              </PrimaryText>
            </Item>
          </CustomBorderView>
          <ActionContainer>
            <PrimaryButton
              {...addAccessibilityLabel('navigateAggregation')}
              title={i18n.t('common.validate')}
              onPress={() => {
                dispatch(getBanksAction());
                navigation.navigate('Aggregation');
              }}
            />
          </ActionContainer>
        </AppBaseView>
      </ScrollView>
    </Root>
  );
};
export default Consentement;
const CustomBorderView = styled(BorderView)`
  align-self: stretch;
`;

const Item = styled.View`
  flex: 1;
  margin-bottom: 15px;
  flex-direction: row;
`;
const IconContainer = styled.View`
  margin-right: 10px;
  margin-top: 6px;
`;
