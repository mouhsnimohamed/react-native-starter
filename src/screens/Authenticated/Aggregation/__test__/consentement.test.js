import { aggregationReducer } from 'modules/aggregation/aggregation.reducer';
import React from 'react';
import { cleanup, render } from 'utils/test/render.utils';
import Consentement from '../Consentement';

afterEach(cleanup);

describe('<Consentement />', () => {
  test('render Consentement correctly', async () => {
    const rendered = render(<Consentement />, {
      reducers: {
        [aggregationReducer]: aggregationReducer,
      },
    }).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
