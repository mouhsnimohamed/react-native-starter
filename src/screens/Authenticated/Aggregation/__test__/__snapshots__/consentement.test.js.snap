// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Consentement /> render Consentement correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <RCTScrollView
    collapsable={false}
    keyboardDismissMode="on-drag"
    keyboardShouldPersistTaps="always"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
  >
    <View>
      <View
        justifyContent="flex-start"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "flexBasis": 0,
              "flexDirection": "column",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "flex-start",
              "paddingBottom": 25,
              "paddingLeft": 25,
              "paddingRight": 25,
              "paddingTop": 25,
            },
          ]
        }
      >
        <Text
          size={16}
          style={
            Array [
              Object {
                "color": "#4A4848",
                "flexShrink": 1,
                "fontSize": 16,
                "fontWeight": "normal",
                "textAlign": "center",
                "textDecorationColor": "black",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "none",
                "width": "100%",
              },
              Object {
                "marginBottom": 20,
                "marginTop": 20,
              },
            ]
          }
        >
          Ce service vous permet de consulter le solde des comptes que vous détenez dans d’autres institutions financières dans votre espace client.
        </Text>
        <View
          pBottom={10}
          pLeft={20}
          pRight={20}
          pTop={20}
          radius={10}
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "borderColor": "#BFCEDE",
                "borderRadius": 10,
                "borderStyle": "solid",
                "borderWidth": 1,
                "marginBottom": 16,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 16,
                "paddingBottom": 10,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 20,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginBottom": 15,
                },
              ]
            }
          >
            <Text
              align="left"
              isPrimary={true}
              semiBold={true}
              style={
                Array [
                  Object {
                    "color": "#003D7C",
                    "flexShrink": 1,
                    "fontFamily": "Segoe UI",
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "textAlign": "left",
                    "textDecorationColor": "black",
                    "textDecorationLine": "none",
                    "textDecorationStyle": "solid",
                    "textTransform": "none",
                    "width": "100%",
                  },
                ]
              }
            >
              Commment ça marche:
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginBottom": 15,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "marginRight": 10,
                    "marginTop": 6,
                  },
                ]
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={8}
                bbWidth={8}
                fill="none"
                focusable={false}
                height={8}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": 8,
                      "width": 8,
                    },
                  ]
                }
                vbHeight={8}
                vbWidth={8}
                width={8}
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M3.075 7.67l3.901-2.888c.588-.434.588-1.137 0-1.572L3.075.323c-.95-.691-2.576-.2-2.576.791v5.764c0 1.004 1.627 1.494 2.576.792z"
                    fill={4290760414}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <Text
              align="left"
              isPrimary={true}
              style={
                Array [
                  Object {
                    "color": "#003D7C",
                    "flexShrink": 1,
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "textAlign": "left",
                    "textDecorationColor": "black",
                    "textDecorationLine": "none",
                    "textDecorationStyle": "solid",
                    "textTransform": "none",
                    "width": "100%",
                  },
                ]
              }
            >
              Choisissez votre compte externe à agréger
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginBottom": 15,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "marginRight": 10,
                    "marginTop": 6,
                  },
                ]
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={8}
                bbWidth={8}
                fill="none"
                focusable={false}
                height={8}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": 8,
                      "width": 8,
                    },
                  ]
                }
                vbHeight={8}
                vbWidth={8}
                width={8}
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M3.075 7.67l3.901-2.888c.588-.434.588-1.137 0-1.572L3.075.323c-.95-.691-2.576-.2-2.576.791v5.764c0 1.004 1.627 1.494 2.576.792z"
                    fill={4290760414}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <Text
              align="left"
              isPrimary={true}
              style={
                Array [
                  Object {
                    "color": "#003D7C",
                    "flexShrink": 1,
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "textAlign": "left",
                    "textDecorationColor": "black",
                    "textDecorationLine": "none",
                    "textDecorationStyle": "solid",
                    "textTransform": "none",
                    "width": "100%",
                  },
                ]
              }
            >
              Authentifiez-vous auprès de votre banque externe
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginBottom": 15,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "marginRight": 10,
                    "marginTop": 6,
                  },
                ]
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={8}
                bbWidth={8}
                fill="none"
                focusable={false}
                height={8}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": 8,
                      "width": 8,
                    },
                  ]
                }
                vbHeight={8}
                vbWidth={8}
                width={8}
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M3.075 7.67l3.901-2.888c.588-.434.588-1.137 0-1.572L3.075.323c-.95-.691-2.576-.2-2.576.791v5.764c0 1.004 1.627 1.494 2.576.792z"
                    fill={4290760414}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <Text
              align="left"
              isPrimary={true}
              style={
                Array [
                  Object {
                    "color": "#003D7C",
                    "flexShrink": 1,
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "textAlign": "left",
                    "textDecorationColor": "black",
                    "textDecorationLine": "none",
                    "textDecorationStyle": "solid",
                    "textTransform": "none",
                    "width": "100%",
                  },
                ]
              }
            >
              Visualisez une  confirmation d’agrégation de comptes.
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "marginBottom": 0,
                "marginTop": 30,
              },
            ]
          }
        >
          <View
            accessibilityLabel="navigateAggregation"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="navigateAggregation"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#003D7C",
                    "borderColor": "black",
                    "borderRadius": 5,
                    "borderStyle": "solid",
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "height": 44,
                    "justifyContent": "center",
                    "marginBottom": 10,
                    "opacity": 1,
                  },
                ]
              }
              testID="navigateAggregation"
              variant="default"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Segoe UI",
                      "fontSize": 16,
                      "paddingBottom": 10,
                      "paddingLeft": 15,
                      "paddingRight": 15,
                      "paddingTop": 10,
                      "textAlign": "center",
                    },
                  ]
                }
                variant="default"
              >
                Valider
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
